### ISSUE TITLE
Command Toggle On/Off for MODs

### ISSUE THREAD

**Issue by PrajjwalDatir** [Synthesized-Infinity/Whatsapp-Botto-Void#15]
Requesting a feature to Toggle on/off a command of the bot.
This command can only be used by MODS of the respective BOT

Way to achieve this : My approach 
Add one more parameter to the constructor
```javascript
super(client, handler, {
            command: 'command_goes_here',
            description: 'command description',
            category: 'category',
            usage: `${client.config.prefix}command`,
            isAvailable : true
        }
```
- will this approach work?

- And where all do I need to make changes?
 src/lib/BaseCommand.ts
 all commands constructor
 anywhere else?

**Comment by AlenSaito1** [comment]
Interesting. 

It's possible by doing the following.

1. Add is `isAvailable` field  in the `config` of `ICommand` interface as an optional boolean https://github.com/Synthesized-Infinity/Whatsapp-Botto-Void/blob/b0c91a3e9406ed7289d7ce684aeaf28e318ab9b1/src/typings/command.d.ts#L5-L20
2. Create a command to change the value of `IsAvailable` to true or false depending on the situation with the run method as something like the following
```TS
// `enable.ts`
run = async (M: ISimplifedMessage, { joined }: IParsedArgs) => {
     //Check if the user is a MOD
     if (.....
     // then get the command from command set
     const cmd = joined.trim().toLowerCase()
     const command = this.handler.commands.get(cmd) ||  this.handler.aliases.get(cmd)
     // if the command is valid
     if (command) {
         command['isAvailable'] = true
         this.handler.commands.set(command.command,command)
         command?.aliases?.forEach((alias) => this.handler.aliases.set(alias, command))
         return void M.reply(`${cmd} is available now`)
     }
    M.reply('Invalid Command')
}
```
Create one for disabling commands too

3. Finally, add an if statement to check if the command is available in the [`handleMessage`](https://github.com/Synthesized-Infinity/Whatsapp-Botto-Void/blob/b0c91a3e9406ed7289d7ce684aeaf28e318ab9b1/src/Handlers/MessageHandler.ts#L12) method of [`MessageHandler`](https://github.com/Synthesized-Infinity/Whatsapp-Botto-Void/blob/b0c91a3e9406ed7289d7ce684aeaf28e318ab9b1/src/Handlers/MessageHandler.ts)
```TS
if (!command.isAvailable) return void M.reply('This command is disabled')
```
I'll try to implement this in the future + PRs are welcome.
Thanks!

**Comment by PrajjwalDatir** [comment]
Thanks!

As we discussed, this states of the variable will disappear, once new process starts.
Need to find a better approach

**Comment by PrajjwalDatir** [comment]
So as we discussed @AlenSaito1 ,
we'll make one set in DB to globally disable/enable command for respective bot.

commenting this to keep track of it.

I guess you will do it...

**Comment by PrajjwalDatir** [comment]
Solved with e8efd45ef4b4625cce1f82a2b64c0112460cb338

closing the issue.
